### PR TITLE
Add TypeScript declarations to iron-autogrow-textarea.

### PR DESCRIPTION
**Pull request by aomarks**
This PR adds TypeScript declarations generated by https://github.com/Polymer/gen-typescript-declarations/

These declarations can be re-generated by running `npm run update-types`.

Tracker: https://github.com/Polymer/gen-typescript-declarations/issues/79

### CODE REVIEW

File: iron-autogrow-textarea.d.ts [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+
+  /**
+   * Returns the underlying textarea.
+   */
+  readonly textarea: any;
+
+  /**
+   * Returns textarea's selection start.
+   */
+  selectionStart: any;
```
Comment (TimvdLippe) [inline iron-autogrow-textarea.d.ts]: Should be number, but jsdoc is probably broken

File: iron-autogrow-textarea.d.ts [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+
+  /**
+   * Returns textarea's selection end.
+   */
+  selectionEnd: any;
```
Comment (TimvdLippe) [inline iron-autogrow-textarea.d.ts]: Same number issue

File: iron-autogrow-textarea.d.ts (hunk 1 above)
Comment (aomarks) [reply iron-autogrow-textarea.d.ts]: That's because we've been annotating getter types with `@type`, but the correct annotation is `@return`.

Comment (aomarks) [reply iron-autogrow-textarea.d.ts]: Fixed.

File: iron-autogrow-textarea.d.ts (hunk 2 above)
Comment (aomarks) [reply iron-autogrow-textarea.d.ts]: Fixed.

**Pull request merged**